src/varpubs/hgvs_extractor.py

```diff
@@ -0,0 +1,77 @@
+import re
+from cyvcf2 import VCF
+
+
+def get_hgvsp_index(vcf: VCF) -> int:
+    """
+    Finds the index of the 'HGVS.p' field in the ANN (annotation) column of the VCF header.
+
+    Args:
+        vcf (VCF): A parsed VCF object from cyvcf2.
+
+    Returns:
+        int: The index position of the 'HGVS.p' field in ANN entries.
+
+    Raises:
+        RuntimeError: If 'ANN' or 'HGVS.p' is not found in the VCF header.
+    """
+    for rec in vcf.header_iter():
+        info = rec.info()
+        if info.get("ID") == "ANN":
+            desc = info.get("Description", "")
+            if "HGVS.p" in desc:
+                fields_str = desc.split("': '")[-1].rstrip("'")
+                fields = [f.strip() for f in fields_str.split("|")]
+                try:
+                    return fields.index("HGVS.p")
+                except ValueError as e:
+                    raise RuntimeError("HGVS.p not found in ANN header") from e
+    raise RuntimeError("ANN field not found in VCF header")
+
+
+def extract_hgvsp_from_vcf(vcf_path: str) -> list[str]:
+    """
+    Extracts all unique HGVS.p terms from the given VCF file and converts them into two formats:
+    - Full form (e.g., 'KRAS p.Gly12Cys')
+    - Short form (e.g., 'KRAS G12')
+
+    Filters out synonymous (silent) mutations.
+
+    Args:
+        vcf_path (str): Path to the annotated VCF file.
+
+    Returns:
+        list[str]: A sorted list of unique variant terms to be searched in PubMed.
+    """
+    vcf = VCF(vcf_path)
+    hgvsp_index = get_hgvsp_index(vcf)
+    gene_index = 3  # Fixed index for gene symbol in ANN field
```

Comment:
I dont think that this is fixed. Its probably best to modify `get_hgvsp_index` to be able to retrieve any index depending on a parameter it takes. Then you can call it once with hgvsp and once for Gene to get the indexes.